runelite-client/src/main/java/net/runelite/client/ui/ContainableFrame.java

```diff
@@ -25,14 +25,126 @@
 package net.runelite.client.ui;
 
 import java.awt.Rectangle;
+import java.awt.event.ComponentAdapter;
+import java.awt.event.ComponentEvent;
+import java.time.Duration;
+import java.time.Instant;
 import javax.swing.JFrame;
 import lombok.Getter;
 
 public class ContainableFrame extends JFrame
 {
+	private static final int LARGE_WIDTH_THRESHOLD = 1600;
+	private static final int NEAR_EDGE_DISTANCE = 50;
+	private static final int PANEL_EXPANDED_WIDTH = PluginPanel.PANEL_WIDTH + PluginPanel.SCROLLBAR_WIDTH;
+
 	@Getter
 	private boolean containedInScreen;
 
+	private boolean expanded;
+	private Rectangle oldBounds;
```

Comment:
I don't think storing oldBounds is good solution, this needs to be done in more generic way.